android/KMEA/app/src/main/java/com/tavultesoft/kmea/KMManager.java

```diff
@@ -412,82 +429,95 @@ private static void copyFile(InputStream inputStream, OutputStream outputStream)
    *
    * Assumption: No legacy keyboards exist in /packages/*.js
    * @param context
    */
   private static void migrateOldKeyboardFiles(Context context) {
-    String legacyLanguagesPath = context.getDir("data", Context.MODE_PRIVATE) + File.separator +
+    String legacyLanguagesPath = context.getDir("data", Context.MODE_PRIVATE).toString() + File.separator +
       KMDefault_LegacyAssetLanguages + File.separator;
-    String legacyFontsPath = context.getDir("data", Context.MODE_PRIVATE) + File.separator +
+    String legacyFontsPath = context.getDir("data", Context.MODE_PRIVATE).toString() + File.separator +
       KMDefault_LegacyAssetFonts + File.separator;
     File legacyLanguagesDir = new File(legacyLanguagesPath);
     File legacyFontsDir = new File(legacyFontsPath);
     if (!legacyLanguagesDir.exists() && !legacyFontsDir.exists()) {
       return;
     }
 
-    String migratedPath = context.getDir("data", Context.MODE_PRIVATE) + File.separator +
-      KMDefault_AssetPackages + File.separator + KMDefault_LegacyPackageID;
+    String migratedPath = context.getDir("data", Context.MODE_PRIVATE).toString() + File.separator +
+      KMDefault_UndefinedPackageID;
     File migratedDir = new File(migratedPath);
     if (!migratedDir.exists()) {
       migratedDir.mkdir();
     }
 
-    String[] files = legacyLanguagesDir.list();
-    if (files != null) {
-      for (String filename : files) {
-        // Handle keyboards missing version number
-        if (filename.lastIndexOf("-") < 0) {
-          if (filename.equals("us.js")) {
-            File kbFile = new File(legacyLanguagesPath, filename);
-            kbFile.delete();
-          } else if (filename.endsWith(".js")) {
-            // Append default version number to keyboard and move
-            String newFilename = filename.substring(0, filename.lastIndexOf(".js")) + "-1.0.js";
-            File kbFile = new File(legacyLanguagesPath, filename);
-            kbFile.renameTo(new File(migratedDir, newFilename));
+    try {
+      if (legacyLanguagesDir.exists()) {
+        FileFilter keyboardFilter = new FileFilter() {
+          @Override
+          public boolean accept(File pathname) {
+            if (pathname.isFile() && pathname.getName().endsWith(".js")) {
+              return true;
+            }
+            return false;
           }
-          // Handle keyboards with version number
-        } else {
-          // Remove legacy default keyboards
-          if (filename.startsWith(KMDefault_KeyboardID + "-")) {
-            File legacyKbFile = new File(legacyLanguagesDir, filename);
-            if (legacyKbFile.exists()) {
-              legacyKbFile.delete();
+        };
+
+        File[] files = legacyLanguagesDir.listFiles(keyboardFilter);
+        for (File file : files) {
+          String filename = file.getName();
+
+          // Keyboards can have filenames "keyboardID.js" or "keyboardID-version.js"
```

Comment:
I'm a bit puzzled by this code sequence below. The legacy keyboards all have version number, don't they? And are we really assuming v1.0 if no version number is appended?

Reply:
The block of code traces back to `renameOldKeyboardFiles()` in the original `release-android-stable-2.8` branch.
I don't know how feasible it was for people to end up with legacy keyboards without version numbers, but it appears this code guarantees it'll at least have `-1.0`.

Reply:
I'm guessing then that this was quite an old piece of code. It's probably no longer needed but we can leave it for now.